examples/vision/metric_learning.py

```diff
@@ -0,0 +1,316 @@
+"""
+Title: Metric learning using crossentropy
+Author: [Mat Kelcey](https://twitter.com/mat_kelcey)
+Date created: 2020/06/05
+Last modified: 2020/06/05
+Description: Example of using metric learning using crossentropy on synthetic data.
+"""
+"""
+## Overview
+
+Metric learning aims to train models that can embed inputs into a high-dimensional space
+such that "similar" inputs, as defined by the training scheme, are located close to each
+other. These models once trained can produce embeddings for downstream systems where such
+similarity is useful; examples include as a ranking signal for search or as a form of
+pretrained embedding model for another supervised problem.
+
+For a more detailed overview of metric learning see:
+
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+"""
+
+import random
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+from collections import defaultdict
+from PIL import Image
+from sklearn.metrics import ConfusionMatrixDisplay
+from tensorflow import keras
+from tensorflow.keras import layers
+
+"""
+## Dataset
+
+For this example we will be using the
+[CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset.
+"""
+
+from tensorflow.keras.datasets import cifar10
+
+(x_train, y_train), (x_test, y_test) = cifar10.load_data()
+
+x_train = x_train.astype("float32") / 255.0
+y_train = np.squeeze(y_train)
```

Comment:
Why the squeeze? There should be no empty dimensions

Reply:
`y_train` is `(50000, 1)` & `y_test` is `(10000, 1)`. just want `(50000,)` to make the enumerate later a bit easier.